### PR TITLE
[OGUI-1581] Split action for taking/releasing all locks in API router

### PR DESCRIPTION
**Pull request by pepijndik**
#### I have JIRA issue created
- [x] branch and/or PR name(s) includes JIRA ID
- [x] issue has "Fix version" assigned
- [x] issue "Status" is set to "In review"
- [x] PR labels are selected
- [x] FLP integration tests were ran successful

Added middleware to append detectorid, and endpoint for action of all detectors

### CODE REVIEW

File: Control/lib/api.js [hunk 1]
```diff
@@ -189,6 +191,13 @@ module.exports.setup = (http, ws) => {
 
   // Lock Service
   http.get('/locks', lockController.getLocksStateHandler.bind(lockController));
+
+  http.put('/locks/:action/ALL',
+    minimumRoleMiddleware(Role.GLOBAL),
+    addDetectorIdMiddleware('ALL'),
```
Comment (graduta) [inline Control/lib/api.js]: We should aim not to use hardcoded strings. In this case, `ALL`
The same one is used in `LockController.jsL22`. Thus, I would like you to create an enum similar to `detectorLockAction.enum.js` with this value and reuse it in both places.

File: Control/lib/middleware/addDetectorId.middleware.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+ * Middleware function to add detectorID to the request object
+ * @return {function(req, res, next): void} - middleware function
+ */
+const addDetectorIdMiddleware = (detectorId) => {
+  return async (req, res, next) => {
+    req.params.detectorId = detectorId;
+    next();
+  };
+}
+exports.addDetectorIdMiddleware = addDetectorIdMiddleware
```
Comment (graduta) [inline Control/lib/middleware/addDetectorId.middleware.js]: Please do not forget to add a new line at the end of the file

File: Control/lib/middleware/addDetectorId.middleware.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+/**
+ * Middleware function to add detectorID to the request object
+ * @return {function(req, res, next): void} - middleware function
+ */
+const addDetectorIdMiddleware = (detectorId) => {
+  return async (req, res, next) => {
```
Comment (graduta) [inline Control/lib/middleware/addDetectorId.middleware.js:23]: You are missing some documentation from the return function. Please have a look `minimumRole.middleware.js` and try to improve the one here as well

File: Control/lib/api.js (hunk 1 above)
Comment (graduta) [reply Control/lib/api.js]: You could also reuse the `ALL` in the route as well once you extract the variable

**Review by graduta — CHANGES_REQUESTED** [review]
Hi @pepijndik !
Very nice and quick PR, especially well done on the testing side!
I added a few minor comments for you to have a look and fix

File: Control/lib/api.js [hunk 4]
```diff
@@ -19,6 +19,8 @@ const config = require('./config/configProvider.js');
 
 // middleware
 const {minimumRoleMiddleware} = require('./middleware/minimumRole.middleware.js');
+const {addDetectorIdMiddleware} = require('./middleware/addDetectorId.middleware.js');
+const {DetectorIdState} = require('./common/lock/detectorIdState.enum.js');
```
Comment (graduta) [inline Control/lib/api.js]: Hi @pepijndik , what is the reason for the `DetectorId` to include `State`?

For example in the test files you refer to it `DetectorId` https://github.com/AliceO2Group/WebUi/pull/2672/files#diff-dd46489f0cf6038851352a0dcca80206a492f27d97623897f7b65f867f4fb35eR19

File: Control/lib/common/lock/detectorIdState.enum.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+ * This software is distributed under the terms of the GNU General Public
+ * License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+ *
+ * In applying this license CERN does not waive the privileges and immunities
+ * granted to it by virtue of its status as an Intergovernmental Organization
+ * or submit itself to any jurisdiction.
+*/
+
+/**
+ * Available lock states
```
Comment (graduta) [inline Control/lib/common/lock/detectorIdState.enum.js]: When copy-ing paste-ing, please make sure what you copy is relevant

File: Control/lib/middleware/addDetectorId.middleware.js (hunk 2 above)
Comment (graduta) [reply Control/lib/middleware/addDetectorId.middleware.js]: @pepijndik , you marked the conversation as resolved but there is still no end of line for the file

**Review by graduta — CHANGES_REQUESTED** [review]

File: Control/lib/api.js (hunk 4 above)
Comment (pepijndik) [reply Control/lib/api.js]: Fixed  this in New Commit

File: Control/test/lib/middleware/mocha-addDetectorId.middleware.test.js [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+
+    addDetectorIdMiddleware(detectorId)(req, res, next);
+
+    // Validate the detectorId is overwritten
+    assert.strictEqual(req.params.detectorId, detectorId);
+    assert.ok(next.calledOnce);
+  });
+
+  it('should integrate with an Express route and add detectorId', async () => {
+    const app = express();
```
Comment (graduta) [inline Control/test/lib/middleware/mocha-addDetectorId.middleware.test.js]: This test is not in-line with the standards of the GUI testing. In this file, we should have only tests for the middleware function and not integrated in a server. 
Moreover, the server is just a usual one and not even the WebUI one. For API integrated services with have a separate `test/api` folder where we test the entire routes. Such tests should go in there and not here.
Could you please remove this test for now and when it comes to API testing we will add it there in a different PR

**Review by graduta — CHANGES_REQUESTED** [review]

File: Control/test/lib/middleware/mocha-addDetectorId.middleware.test.js (hunk 6 above)
Comment (pepijndik) [reply Control/test/lib/middleware/mocha-addDetectorId.middleware.test.js]: oké will remove it,

File: Control/test/api/lock/api-put-locks.test.js [hunk 7]
```diff
@@ -164,4 +164,35 @@ describe(`'API - PUT - /locks/:action/:detectorId' test suite`, () => {
         ODC: { name: 'ODC', state: 'FREE' },
       });
   });
+
+  it('should successfully Take ALL locks when >=global ', async () => {
+    // first we retake a lock to ensure we have a lock to release from different types of users
```
Comment (graduta) [inline Control/test/api/lock/api-put-locks.test.js]: I believe this comment is not related to the test and following lines?

**Review by graduta — CHANGES_REQUESTED** [review]
Please have a look at one comment I added but also at the GitHub Code Scanning comments

**Review by graduta — APPROVED** [review]
